tests/test_lambda_dispatch.py

```diff
@@ -0,0 +1,175 @@
+                },
+            }
+        ]
+    }
+    yield sqs_msg
+
+@pytest.fixture
+def event(s3_event, sns_message_template, sqs_message_template):
+    sns_message_template["Message"] = json.dumps({"Records": [s3_event]})
+    sqs_message_template["Records"][0]["body"] = json.dumps(sns_message_template)
+    return sqs_message_template
+
+@pytest.fixture
+def archive_json_paths():
+    archive_json_paths = [
+            "HealthKitV2Workouts_20240508-20240509.json", # normal file
+            "empty.json", # should have file size 0 and be ignored
+            "Manifest.json", # should be ignored
+            "dir/containing/stuff.json" # should be ignored
+    ]
+    return archive_json_paths
+
+@pytest.fixture
+def temp_zip_file():
+    temp_zip_file = tempfile.NamedTemporaryFile(
+            delete=False,
+            suffix='.zip'
+    )
+    return temp_zip_file
+
+@pytest.fixture
+def archive_path(archive_json_paths, temp_zip_file):
+    with zipfile.ZipFile(temp_zip_file.name, 'w', zipfile.ZIP_DEFLATED) as zip_file:
+        for file_path in archive_json_paths:
+            if "/" in file_path:
+                os.makedirs(os.path.dirname(file_path))
+            if file_path != "empty.json":
+                with open(file_path, "w") as this_file:
+                    this_file.write("test")
+            else:
+                Path(file_path).touch()
+            zip_file.write(file_path)
+            if "/" in file_path:
+                shutil.rmtree(file_path.split("/")[0])
+            else:
+                os.remove(file_path)
+    yield temp_zip_file.name
+    os.remove(temp_zip_file.name)
+
+def test_get_object_info(s3_event):
+    object_info = app.get_object_info(s3_event=s3_event)
+    assert object_info["Bucket"] == s3_event["s3"]["bucket"]["name"]
+    assert object_info["Key"] == s3_event["s3"]["object"]["key"]
+
+def test_get_object_info_unicode_characters_in_key(s3_event):
+    s3_event["s3"]["object"]["key"] = \
+            "main/2023-09-26T00%3A06%3A39Z_d873eafb-554f-4f8a-9e61-cdbcb7de07eb"
+    object_info = app.get_object_info(s3_event=s3_event)
+    assert object_info["Key"] == \
+            "main/2023-09-26T00:06:39Z_d873eafb-554f-4f8a-9e61-cdbcb7de07eb"
+
+def test_get_archive_contents(archive_path, archive_json_paths):
+    dummy_bucket = "dummy_bucket"
+    dummy_key = "dummy_key"
+    archive_contents = app.get_archive_contents(
+            archive_path=archive_path,
+            bucket=dummy_bucket,
+            key=dummy_key
+    )
+    assert all([content["Bucket"] == dummy_bucket for content in archive_contents])
+    assert all([content["Key"] == dummy_key for content in archive_contents])
+    assert all([content["FileSize"] > 0 for content in archive_contents])
+    assert set([content["Path"] for content in archive_contents]) == \
+            set(["HealthKitV2Workouts_20240508-20240509.json"])
+
+@mock_sns
+@mock_s3
+def test_main(event, temp_zip_file, s3_event, archive_path):
```

Comment:
Should we have a test case testing that the sns doesn't get called if archive contents comes up empty?

Reply:
I don't think that's necessary.

```
archive_contents = get_archive_contents(...)
for file_info in archive_contents:
    # do stuff
    sns_client.publish(...)
```
Since the logic is straightforward and the only time we call `publish` is inside this for loop, it's overkill to check whether the for loop never runs its block if the `archive_contents` list is empty.